src/plugin.rs

```diff
@@ -316,6 +374,17 @@ impl Instance {
                 .connect_port_mut(index.0, data.as_mut_ptr());
         }
         self.inner.run(ports.sample_count);
+
+        let worker_interface = worker::maybe_get_worker_interface(&mut self.inner);
```

Comment:
Can you store the result of line 103 in the struct instead of calling `worker::maybe_get_worker_interface` again? Asking because worker::maybe_get_worker_interface contains an allocation (through `CString::new`) somewhere down the stack. We could also tweak the code to take a preconstructed `CStr`, but storing the result may be more efficient anyways.

Then you would do `if let Some(interface) = self.worker_interface.as_mut() { ... }`

Reply:
Done, thank you I didn't realize there was an allocation happening there. Out of interest how do you find where allocations are happening other than digging through the code?

Reply:
I read the code with the help of rust-analyzer.

Seems like the crate [assert_no_alloc](https://docs.rs/assert_no_alloc/latest/assert_no_alloc/) can be used for testing allocations on the Rust side.